File: reset_desc.py
import json

CATALOG_FILE = 'assets/js/catalog.js'
PRODUCTS_JSON_FILE = 'scratch_products.json'

with open(CATALOG_FILE, 'r', encoding='utf-8') as f:
    content = f.read()

start_marker = "var PRODUCTS = ["
end_marker = "];\n\n// ============================================"

start_idx = content.find(start_marker)
end_idx = content.find(end_marker)

if start_idx == -1 or end_idx == -1:
    print("Could not find PRODUCTS array.")
    exit(1)

# Extract just the JSON array part
# start_idx + 15 is the '['
# end_idx is the ']'
products_json_str = content[start_idx + 15 : end_idx + 1]

products = json.loads(products_json_str)

with open(PRODUCTS_JSON_FILE, 'r', encoding='utf-8') as f:
    original_products = json.load(f)

orig_desc_map = {p['id']: p.get('desc', '') for p in original_products}

changes = 0
for p in products:
    pid = p['id']
    orig_desc = orig_desc_map.get(pid, '')
    
    if orig_desc:
        p['desc'] = orig_desc
    else:
        p['desc'] = f"Код 1С: {pid}"
        
    changes += 1

print(f"Reset descriptions for {changes} products.")

new_products_str = json.dumps(products, ensure_ascii=False, indent=2)

# Now safely assemble the new content
# content[:start_idx + 15] contains 'var PRODUCTS = ['
# new_products_str has the JSON array [...]
# BUT we must remove the first '[' and last ']' from new_products_str to fit into the marker?
# NO, new_products_str is the ENTIRE array.
# so we replace everything between 'var PRODUCTS = ' and ';\n\n// ==', meaning we replace `content[start_idx + 15 : end_idx + 1]` with `new_products_str`.

new_content = content[:start_idx + 15] + new_products_str + content[end_idx + 1:]

with open(CATALOG_FILE, 'w', encoding='utf-8') as f:
    f.write(new_content)

print("catalog.js successfully updated with clean descriptions!")

File: reset_desc.js
const fs = require('fs');

const catalogPath = 'assets/js/catalog.js';
const origPath = 'scratch_products.json';

let content = fs.readFileSync(catalogPath, 'utf8');

// We need to parse PRODUCTS from catalog.js
// It's a valid JS file. Let's just run it in a new context or use eval (safe here since we generated it).
let sandbox = {};
const vm = require('vm');
vm.createContext(sandbox);
vm.runInContext(content, sandbox);

let products = sandbox.PRODUCTS;
let origProducts = JSON.parse(fs.readFileSync(origPath, 'utf8'));

let origMap = {};
for (let p of origProducts) {
    origMap[p.id] = p.desc || '';
}

let changes = 0;
for (let p of products) {
    let origDesc = origMap[p.id];
    if (origDesc) {
        p.desc = origDesc;
    } else {
        p.desc = `Код 1С: ${p.id}`;
    }
    changes++;
}

console.log(`Updated ${changes} descriptions.`);

// Now we need to replace the PRODUCTS array in the string
// Since we generated it with JSON.stringify, we can just replace the block between `var PRODUCTS = [` and `];\n\n// ====`
let startMarker = 'var PRODUCTS = ';
let endMarker = '];\n\n// ============================================';

let startIdx = content.indexOf(startMarker);
let endIdx = content.indexOf(endMarker);

if (startIdx !== -1 && endIdx !== -1) {
    let newJson = JSON.stringify(products, null, 2);
    let newContent = content.substring(0, startIdx + startMarker.length) + newJson + content.substring(endIdx + 1);
    fs.writeFileSync(catalogPath, newContent, 'utf8');
    console.log("Successfully updated catalog.js");
} else {
    console.log("Could not find markers.");
}
